applications/models.py

```diff
@@ -294,6 +307,20 @@ class ApplicationRoundBasket(models.Model):
         verbose_name=_("Order number"), default=1, null=False, blank=True
     )
 
+    def get_application_events_in_basket(self, excluded_events):
+        events = ApplicationEvent.objects.filter(
+            application__application_round=self.application_round
+        )
+
+        if self.purpose is not None:
+            events = events.filter(purpose=self.purpose)
+        if len(self.age_groups.all()) > 0:
+            events = events.filter(age_group__in=self.age_groups.all())
+        if len(excluded_events) > 0:
+            events = events.exclude(id__in=map(lambda x: x.id, excluded_events))
```

Comment:
Not a change request; this could also be written as: `events = events.exclude(id__in=excluded_events.values_list("id", flat=True))` if the excluded_events is a queryset.